### PR TITLE
feat: add more ignored files

### LINKED ISSUE #869: 📎 Add more Known files to ignore
### Description

Biome ignores below files (you can check [here on the website](https://biomejs.dev/guides/how-biome-works/#known-files)):

- package.json
- package-lock.json
- tsconfig.json
- jsconfig.json

We can add more files to ignore, for example: 

- ~bun.lockb~
- ~package-lock.yaml~
- deno.json (should we also ignore deno.jsonc?)

ref: https://github.com/biomejs/biome/discussions/840
Thanks for bringing this up @jacobthesheep

### PR DESCRIPTION
**Pull request by Gumichocopengin8**
## Summary
fix #869
Let me know if I need to add/remove (more) file(s)

## Test Plan
- Check if newly added files are applicable to ignore formatting
- Check if all ignored files are listed in `/guides/how-biome-works/#known-files` (`pnpm start` in `website` dir)

### CODE REVIEW

File: crates/biome_fs/src/fs/os.rs [hunk 1]
```diff
@@ -244,9 +244,21 @@ fn handle_dir_entry<'scope>(
             Some(
                 "package.json"
                     | "package-lock.json"
+                    | "package-lock.yaml"
+                    | ".npmrc"
+                    | "npm-shrinkwrap.json"
+                    | "yarn.lock"
+                    | "pnpm-lock.yaml"
+                    | "pnpm-workspace.yaml"
+                    | "composer.json"
```
Comment (ematipico) [inline crates/biome_fs/src/fs/os.rs:253]: Out of curiosity, which does create this file?

**Review by ematipico — APPROVED** [review]

File: crates/biome_fs/src/fs/os.rs (hunk 1 above)
Comment (unvalley) [reply crates/biome_fs/src/fs/os.rs:253]: `composer.json` maybe created on php project.

Comment (Gumichocopengin8) [reply crates/biome_fs/src/fs/os.rs:253]: - `yarn.lock` is for `yarn`
- `pnpm-lock.yaml` and `pnpm-workspace.yaml` for `pnpm`
- `composer.json` is for PHP

Does that answer your question?

**Review by unvalley — APPROVED** [review]
We may as well turn off everything but json-like since we don't emit anything but json for now.

**Comment by Gumichocopengin8** [comment]
> Thanks! We may as well turn off everything but json-like since we don't emit anything but json for now. We can remove yaml and toml for now.

Let me clarify. The ticket also mentions `bun.lockb`. We wanna remove `bun.lockb` as well as non-json files since it's a binary file. Is it correct?

**Comment by unvalley** [comment]
@Gumichocopengin8 I confuse them. Let's remove that.

File: crates/biome_fs/src/fs/os.rs (hunk 1 above)
Comment (nhedger) [reply crates/biome_fs/src/fs/os.rs:253]: `composer.json` is the equivalent of `packages.json` in the PHP world. It is managed by [Composer](https://getcomposer.org)

**Comment by Gumichocopengin8** [comment]
@unvalley done!

File: crates/biome_fs/src/fs/os.rs [hunk 2]
```diff
@@ -244,9 +244,21 @@ fn handle_dir_entry<'scope>(
...
+                    | "composer.lock"
                     | "typescript.json"
                     | "tsconfig.json"
                     | "jsconfig.json"
+                    | "bun.lockb"
+                    | "bunfig.toml"
```
Comment (unvalley) [inline crates/biome_fs/src/fs/os.rs:259]:
```suggestion
                    | "npm-shrinkwrap.json"
                    | "yarn.lock"
                    | "composer.json"
                    | "composer.lock"
                    | "typescript.json"
                    | "tsconfig.json"
                    | "jsconfig.json"
```

Comment (Gumichocopengin8) [reply crates/biome_fs/src/fs/os.rs:259]: My bad :( 
The file change wasn't included in the commit

**Pull request merged**